R/app.R

```diff
@@ -186,13 +187,10 @@ with_head_tags <- function(ui) {
 # For Shiny to reload the app correctly, the body of the server function must meet two criteria:
 # 1. It must be wrapped in curly braces.
 # 2. It must have source reference information attached, i.e. `srcref` attributes.
-#
-# (1) is ensured by `normalize_server()`.
-# (2) is ensured by `reparse()`.
-reparse <- function(f) {
-  # Deparse the function and parse it again with `keep.source = TRUE`.
-  eval(
-    expr = parse(text = deparse(f), keep.source = TRUE),
-    envir = environment(f)
-  )
+fix_server <- function(server) {
+  force(server) # Avoid the pitfalls of lazy evaluation.
+  eval(parse(
+    text = "function(input, output, session) { server(input, output, session) }",
```

Comment:
I think that it would be beneficial to store the implementation details and the reasoning somewhere (e.g. GitHub issue that will be closed by this PR).